Netable/Example/Views/RootView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  RootView.swift
+//  Example
+//
+//  Created by Amy Oulton on 2022-12-01.
+//  Copyright © 2022 Steamclock Software. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+struct RootView: View {
+    @ObservedObject var viewModel: RootVM
+
+    var body: some View {
+        VStack {
+            if let error = viewModel.error {
+                VStack {
+                    HStack {
+                        Spacer()
+                        Text("Error: \(error)")
+                        Spacer()
+                        Image(systemName: "xmark")
+                            .foregroundColor(.black)
+                            .onTapGesture {
+                                viewModel.clearError()
+                            }
+                    }
+                }.frame(maxWidth: .infinity)
+                .padding()
+                .background(.yellow)
+                .onAppear {
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
+                        viewModel.error = nil
+                    }
+                }
+            }
+            if viewModel.user == nil {
+                loginView
```

Comment:
Sounds great. I think practically it makes sense to leave the login functionality within the RootVM because of how intertwined it is an immediate it calls the rest of the root functions, but if you think it would be better to separate it I'm happy to do so!